File: tests/fixtures/tslp_corpus/typescript/sample.ts
// Hand-counted TypeScript fixture for the F5 corpus gate.
import type { Readable } from "node:stream";
import { EventEmitter } from "node:events";

export type Pair = [string, number];

export interface Countable {
  count(): number;
}

export enum Outcome {
  Hit,
  Miss,
}

export class Counter extends EventEmitter implements Countable {
  // A class field is not a symbol under this extractor's rule; the manifest's
  // count depends on that.
  private hits = 0;

  count(): number {
    return this.hits;
  }

  bump(): void {
    this.hits += 1;
  }
}

export function build(stream?: Readable): [Counter, Pair, Outcome] {
  void stream;
  return [new Counter(), ["fixture", 1], Outcome.Hit];
}

File: tests/fixtures/tslp_corpus/python/sample.py
"""Hand-counted Python fixture for the F5 corpus gate."""

import os
import sys as system
from collections import OrderedDict
from typing import Iterable, Optional

LIMIT = 8


def top_level(value: int) -> int:
    def inner(x: int) -> int:
        return x + 1

    return inner(value)


class Counter:
    """Counts things."""

    def __init__(self) -> None:
        self.hits = 0

    def bump(self) -> None:
        self.hits += 1

    class Nested:
        def deep(self) -> str:
            return "deep"


async def fetch(url: str) -> Optional[str]:
    _ = (os, system, OrderedDict, Iterable, LIMIT, url)
    return None
